lib/kafka/monitor.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'logging/monitor'
+require 'zero_silent_failures/monitor'
+
+module Kafka
+  # This Monitor class is responsible for tracking and logging various events related to the Kafka service.
+  # It inherits from the ZeroSilentFailures::Monitor class and provides methods to track the
+  # success and failure of submissions.
+  class Monitor < Logging::Monitor
+    # metric prefix
+    STATSD_KEY_PREFIX = 'api.kafka_service'
+
+    def initialize
+      super('kafka-service')
+    end
+
+    # Track submission successful
+    # @see Kafka::EventBusSubmissionJob
+    #
+    # @param topic [String] The Kafka topic to which the message will be sent
+    # @param payload [Hash] The message payload to be sent to the Kafka topic
+    def track_submission_success(topic, payload)
+      payload.dig('data', 'ICN') && (payload['data']['ICN'] = '[REDACTED]')
+
+      additional_context = { topic:, payload: }
+      track_request(
+        'info',
+        "Kafka::EventBusSubmissionJob submission succeeded for topic #{topic}",
```

Comment:
Could we make this more generic so it can be used outside of the Kafka::EventBusSubmissionJob?

Reply:
@aherzberg This specific method is for the submission and is not _really_ supposed to be too general. We can create new methods for new logging to keep things separate. Do you have an example of something else that would use this if we did make it more generic?

Reply:
In that case, maybe we rename the class to EventBusSubmissionJobMonitor? I don't have anything specific in mind.

Reply:
@aherzberg You're losing me a bit. The monitor classes all follow a specific nomenclature if you peak other monitor.rb files as examples.

Reply:
@balexandr could we add an EventBusSubmissionJob namespace then instead? If the monitor needs to be specific, we should be describing it for what its specifically for.

Reply:
@aherzberg Sorry I meant the methods inside the class. All the monitor classes should be called Monitor. You'll notice that it's in `kafka/monitor.rb` which means it is already specific to all things kafka.

Reply:
@balexandr I should have mentioned that we do have plans to create a synchronous option to submit an event so if we want to use this monitor for both cases we could just update the message to "Kafka submission failed for topic #{topic}" and let the call_location define if it's coming from the async job or the synchronous location

Reply:
Historically, we've kept everything in one file per "thing". One for Pensions, one for Burial, etc. If you want to use it for `AvroProducer` or `ProducerManager` you can create methods like `track_avro_producer_success` or `track_producer_manager_success`.

Having it all in one file forces you to use unique method names and makes it easier to read when reading code IMHO

Reply:
Alright, we can leave it as it is.